xla/stream_executor/gpu/all_reduce_kernel_lib.cu.h

```diff
@@ -39,10 +47,10 @@ union alignas(16) Vec<float> {
 };
 
 template <>
-union alignas(8) Vec<__nv_bfloat16> {
+union alignas(8) Vec<bf16_t> {
   using PackedType = int2;
 
-  __nv_bfloat16 data[4];
+  bf16_t data[4];
   PackedType packed;
 };
```

Comment:
To avoid macros can you move this specialization into all_reduce_kernel_cuda.cc and all_reduce_kernel_rocm.cc?